geode-core/src/main/java/org/apache/geode/internal/cache/xmlcache/RegionAttributesCreation.java

```diff
@@ -1550,15 +1551,35 @@ public PartitionAttributes getPartitionAttributes() {
     return this.partitionAttributes;
   }
 
-  public void setPartitionAttributes(PartitionAttributes pa) {
-    if (pa != null) {
+  public void setPartitionAttributes(PartitionAttributes partitionAttr) {
+    if (partitionAttr != null) {
       if (!hasDataPolicy()) {
-        this.setDataPolicy(PartitionedRegionHelper.DEFAULT_DATA_POLICY);
+        setDataPolicy(PartitionedRegionHelper.DEFAULT_DATA_POLICY);
         setHasDataPolicy(false);
+      } else if (!PartitionedRegionHelper.ALLOWED_DATA_POLICIES.contains(getDataPolicy())) {
```

Comment:
Everything else looks like a replacement but this is a new branch of code.  Would you be able to elaborate a bit, just for my own understanding...

Reply:
Thanks for reviewing @jhuynh1.
This branch is basically a copy of the validations we execute in the deprecated `AttributesFactory` class, just to make sure the caller can't set an invalid  data policy for the partition region (only `PARTITION` and `PERSISTENT_PARTITION` are allowed).